server/src/test/YmlKaoFileVisitor.test.ts

```diff
@@ -0,0 +1,247 @@
+import { ANTLRInputStream, CommonTokenStream } from "antlr4ts";
+import * as assert from "assert";
+import { CompletionItem } from "vscode-languageserver";
+import YmlKaoFileVisitor from "../visitors/YmlKaoFileVisitor";
+import { YmlToBdlLexer } from "../YmlToBdlLexer";
+import { YmlToBdlParser } from "../YmlToBdlParser";
+
+describe("Extension Server Tests", () => {
+  describe("YmlKaoFileVisitor", () => {
+    it("should be OK to instantiate YmlKaoFileVisitor", (done) => {
+      const inputStream = new ANTLRInputStream("");
+      const lexer = new YmlToBdlLexer(inputStream);
+      const tokenStream = new CommonTokenStream(lexer);
+      const parser = new YmlToBdlParser(tokenStream);
+
+      const result = parser.kaoFile();
+      const visitor = new YmlKaoFileVisitor([]);
+      visitor.visit(result);
+      done();
+    });
+
+    it("should be able to parse a string", (done) => {
+      const inputStream = new ANTLRInputStream('"foo bar"');
+      const lexer = new YmlToBdlLexer(inputStream);
+      const tokenStream = new CommonTokenStream(lexer);
+      const tokens = tokenStream.getTokens();
+      assert.equal(tokens.length, 1);
+      assert.equal(tokens[0].startIndex, 0);
+      assert.equal(tokens[0].stopIndex, 8);
+      done();
+    });
+
+    it("should parse a well written YML class and provide completion for fields", (done) => {
+      const inputStream = new ANTLRInputStream(`
+            interface City
+                field name
+                --> domains	String
+
+                field country
+                --> domains String
+
+                classProperties
+                --> dsl -> Textualization
+                        --> root "City"
+                        --> language LANG_fr
+                        --> gender FEMININE
+                        --> number SINGULAR
+                        ;
+                --> hasAccessorFunction true
+                ;
+
+                implementation City
+                ;
+            `);
+      const lexer = new YmlToBdlLexer(inputStream);
+      const tokenStream = new CommonTokenStream(lexer);
+      const parser = new YmlToBdlParser(tokenStream);
+
+      const result = parser.kaoFile();
+      const completionItems: CompletionItem[] = [];
+      const visitor = new YmlKaoFileVisitor(completionItems);
+      visitor.visit(result);
+      const expectedCompletionItems = [
+        {
+          data: "id_City_name",
+          detail: "String",
+          documentation: "not documented",
+          kind: 10,
```

Comment:
If the kind is (I might be wrong) an auto-generated Antlr token number or something of this kind that might change, checking it might make your test needlessly brittle. Any opinion on this?